pkg/controller/shared_informer.go

```diff
@@ -0,0 +1,101 @@
+package controller
+
+import (
+	"reflect"
+	"sync"
+	"time"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/watch"
+
+	oclient "github.com/openshift/origin/pkg/client"
+)
+
+type InformerFactory interface {
+	Start(stopCh <-chan struct{})
+
+	Pods() PodInformer
+}
+
+type PodInformer interface {
+	Informer() framework.SharedIndexInformer
+	Indexer() cache.Indexer
```

Comment:
> I would prefer to get these from the informer since it already has them and we don't need to maintain different interfaces. It seems that the underlying controller implementations are responsible for boiling their interfaces down to what they need.

In order to lazily create the informer, we have to have a semantic understanding of the type we're requesting.  It might be possible to avoid this with some config + unique dynamic client shenanigans, but I think we'll want the `Lister()` API to be typed for every sort of object we have.

Are you trying to say that you just want `PodInformer() framework.SharedIndexInformer` method on the `InformerFactory` and we leave to things like admission plugins to go from an Informer down to an Index or Lister?  I'm not necessarily against that, just want to make sure I've got it right.